fpdf/font_type_3.py

```diff
@@ -223,11 +223,15 @@ def __init__(self, fpdf: "FPDF", base_font: "TTFFont"):
             }
         self.palette = None
         if "CPAL" in self.base_font.ttfont:
-            # hardcoding the first palette for now
-            print(
-                f"This font has {len(self.base_font.ttfont['CPAL'].palettes)} palettes"
-            )
-            palette = self.base_font.ttfont["CPAL"].palettes[0]
+            num_palettes = len(self.base_font.ttfont["CPAL"].palettes)
+            # Validate palette index
+            if palette_index >= num_palettes:
+                LOGGER.warning(
+                    f"Palette index {palette_index} is out of range. "
+                    f"This font has {num_palettes} palettes. Using palette 0."
+                )
```

Comment:
```suggestion
                LOGGER.warning(
                    "Palette index %s is out of range. This font has %s palettes. Using palette 0.",
                    palette_index,
                    num_palettes,
                )
```

Please change from f-string to lazy-formatting to comply with pylint.